src/rest.cpp

```diff
@@ -94,9 +96,15 @@ static NodeContext* GetNodeContext(const util::Ref& context, HTTPRequest* req)
  *                 context mempool is not found.
  * @returns        Pointer to the mempool or nullptr if no mempool found.
  */
-static CTxMemPool* GetMemPool(const util::Ref& context, HTTPRequest* req)
+static CTxMemPool* GetMemPool(const std::any& context, HTTPRequest* req)
 {
-    NodeContext* node = context.Has<NodeContext>() ? &context.Get<NodeContext>() : nullptr;
+    const NodeContext* node;
+    try {
+        NodeContext& node_context = std::any_cast<std::reference_wrapper<NodeContext>>(context);
+        node = &node_context;
+    } catch (const std::bad_any_cast&) {
+        node = nullptr;
+    }
```

Comment:
In commit "refactor: replace util::Ref by std::any (C++17)" (c6686a3a582670618b685e69bdfa2ba893b5ce92)

This change and other changes in this commit seems pretty obscure and verbose. It would be nice to define a helper function so these 7 lines could be just be replaced with:

```c++
NodeContext* node = util::AnyPtr<NodeContext>(context);
```

Helper could go in `util/system.h` and look like:

```c++
template<typename T>
T* AnyPtr(const std::any& any)
{
    try {
        return std::any_cast<T*>(any);
    } catch (const std::bad_any_cast&) {
        return nullptr;
    }
}
```

Also I think it would be better to just assign `context = &node` instead of `context = std::ref(node)`. I think the only reason you would use reference_wrappers instead of pointers is when you want non-nullable references, but std::any is inherently nullable so combining it with std::reference_wrapper just makes things complicated for no benefit.

Reply:
Both introducing a `std::any` access helper and using pointers instead of references are great ideas, thanks! Looks much cleaner now, with the clunky try/catch-construct deduplicated and hidden in util/system.h.